service/configure.ac

```diff
@@ -321,8 +321,12 @@ AC_COMPILE_IFELSE([
 
 AC_LANG_POP([C++])
 
-[EXTRA_CFLAGS="$EXTRA_CFLAGS -Wall -Werror"]
-[EXTRA_CXXFLAGS="$EXTRA_CXXFLAGS -Wall -Werror"]
+m4_include([m4/ax_check_compile_flag.m4])
+AX_CHECK_COMPILE_FLAG([-fp-model="fast"], [FASTMATH="-fp-model=fast"], [FASTMATH="-ffast-math"], [], [])
+AX_CHECK_COMPILE_FLAG([-fp-model="precise"], [NO_FASTMATH="-fp-model=precise"], [NO_FASTMATH="-fno-fast-math"], [], [])
+
+[EXTRA_CFLAGS="$EXTRA_CFLAGS -Wall -Werror $NO_FASTMATH"]
+[EXTRA_CXXFLAGS="$EXTRA_CXXFLAGS -Wall -Werror $NO_FASTMATH"]
```

Comment:
Can you verify that [this case](https://github.com/geopm/geopm/issues/2252#issuecomment-1213135326) from #2252 (monitor agent + geopmbench + icpx) produces a valid `geopm.report` file with this PR's current revision?

If that test passes, then everything LGTM. If that test fails, then I think these changes may be needed in the HPC Runtime's configure.ac as well.